pkg/consensus/consensus.go

```diff
@@ -136,18 +140,11 @@ func (c *Consensus) Start() error {
 	c.checkpoint.Set(c.LastProposal, c.LastSignatures)
 
 	c.createComponents()
-	opts := algorithm.PoolOptions{
-		QueueSize:         int64(c.Config.RequestPoolSize),
-		ForwardTimeout:    c.Config.RequestForwardTimeout,
-		ComplainTimeout:   c.Config.RequestComplainTimeout,
-		AutoRemoveTimeout: c.Config.RequestAutoRemoveTimeout,
-		RequestMaxBytes:   c.Config.RequestMaxBytes,
-		SubmitTimeout:     c.Config.RequestPoolSubmitTimeout,
-		Metrics:           c.Metrics.MetricsRequestPool,
-	}
-	c.submittedChan = make(chan struct{}, 1)
-	c.Pool = algorithm.NewPool(c.Logger, c.RequestInspector, c.controller, opts, c.submittedChan)
-	c.continueCreateComponents()
+
+	// TODO use request pool metrics
+	c.Pool = request.NewPool(c.Logger, c, c.createPoolOpts())
```

Comment:
not sure that "this" consensus should be used as the request inspector. 
Why not make a custom struct type that wraps around the `c.RequestInspector` ?